repository/service.go

```diff
@@ -12,42 +12,58 @@ var (
 type (
 	// Store or retrieve repositories from some database.
 	Store interface {
-		ListByOwnerUsername(string) ([]*Repository, []*Stats, *Owner, error)
-		Find(string, string) (*Repository, *Stats, *Owner, error)
-		Create(ownerID string, repository *Repository) (*Repository, error)
+		List(owner *Owner) ([]*Repository, []*Stats, *Owner, error)
+		Find(owner *Owner, name string) (*Repository, *Stats, *Owner, error)
+		Create(owner *Owner, repository *Repository) (*Repository, error)
+	}
+
+	// Storage manages the git storage
+	Storage interface {
+		Create(owner string, name string) error
 	}
 
 	// Service to interact with repositories.
 	Service interface {
-		ListByOwnerUsername(string) ([]*Repository, []*Stats, *Owner, error)
-		Find(string, string) (*Repository, *Stats, *Owner, error)
-		Create(ownerID string, repository *Repository) (*Repository, error)
+		List(owner *Owner) ([]*Repository, []*Stats, *Owner, error)
```

Comment:
these functions are probably best left as you used to have them - taking an *Owner can be really unclear about "what fields need to be set for this function to work", and you can't always work with 100% hydrated models - thoughts?

Reply:
That's a really good point.
The problem is, that for some `func`s I need both, the id and username. In the first `func` called I need the ID and in the seconds one I need the username. So I thought about combining both. Not really sure though. Maybe I might just require them both as strings too.

Reply:
lately I've been favoring DB interaction functions that take and return "primitive" types, like https://github.com/fortytw2/hydrocarbon/blob/master/db.go#L77 - in the cases where structs are returned almost always the structs are 100% hydrated. 

if you need either/or, I would instead have two functions, I think